llvm/lib/SYCLLowerIR/PrepareSYCLNativeCPU.cpp

```diff
@@ -196,14 +197,41 @@ Function *cloneFunctionAndAddParam(Function *OldF, Type *T) {
 }
 
 // Todo: add support for more SPIRV builtins here
-static std::map<std::string, std::string> BuiltinNamesMap{
-    {"__spirv_BuiltInGlobalInvocationId", "__dpcpp_nativecpu_global_id"},
-    {"__spirv_BuiltInGlobalSize", "__dpcpp_nativecpu_global_range"},
-    {"__spirv_BuiltInWorkgroupSize", "__dpcpp_nativecpu_get_wg_size"},
-    {"__spirv_BuiltInWorkgroupId", "__dpcpp_nativecpu_get_wg_id"},
-    {"__spirv_BuiltInLocalInvocationId", "__dpcpp_nativecpu_get_local_id"},
-    {"__spirv_BuiltInNumWorkgroups", "__dpcpp_nativecpu_get_num_groups"},
-    {"__spirv_BuiltInGlobalOffset", "__dpcpp_nativecpu_get_global_offset"}};
+static std::map<std::string, std::pair<std::string, unsigned int>>
+    BuiltinNamesMap{
+        {"_Z28__spirv_GlobalInvocationId_xv",
+         {"__dpcpp_nativecpu_global_id", 0}},
+        {"_Z28__spirv_GlobalInvocationId_yv",
+         {"__dpcpp_nativecpu_global_id", 1}},
+        {"_Z28__spirv_GlobalInvocationId_zv",
+         {"__dpcpp_nativecpu_global_id", 2}},
+        {"_Z20__spirv_GlobalSize_xv", {"__dpcpp_nativecpu_global_range", 0}},
+        {"_Z20__spirv_GlobalSize_yv", {"__dpcpp_nativecpu_global_range", 1}},
+        {"_Z20__spirv_GlobalSize_zv", {"__dpcpp_nativecpu_global_range", 2}},
+        {"_Z22__spirv_GlobalOffset_xv",
+         {"__dpcpp_nativecpu_get_global_offset", 0}},
+        {"_Z22__spirv_GlobalOffset_yv",
+         {"__dpcpp_nativecpu_get_global_offset", 1}},
+        {"_Z22__spirv_GlobalOffset_zv",
+         {"__dpcpp_nativecpu_get_global_offset", 2}},
+        {"_Z27__spirv_LocalInvocationId_xv",
+         {"__dpcpp_nativecpu_get_local_id", 0}},
+        {"_Z27__spirv_LocalInvocationId_yv",
+         {"__dpcpp_nativecpu_get_local_id", 1}},
+        {"_Z27__spirv_LocalInvocationId_zv",
+         {"__dpcpp_nativecpu_get_local_id", 2}},
+        {"_Z23__spirv_NumWorkgroups_xv",
+         {"__dpcpp_nativecpu_get_num_groups", 0}},
+        {"_Z23__spirv_NumWorkgroups_yv",
+         {"__dpcpp_nativecpu_get_num_groups", 1}},
+        {"_Z23__spirv_NumWorkgroups_zv",
+         {"__dpcpp_nativecpu_get_num_groups", 2}},
+        {"_Z23__spirv_WorkgroupSize_xv", {"__dpcpp_nativecpu_get_wg_size", 0}},
+        {"_Z23__spirv_WorkgroupSize_yv", {"__dpcpp_nativecpu_get_wg_size", 1}},
+        {"_Z23__spirv_WorkgroupSize_zv", {"__dpcpp_nativecpu_get_wg_size", 2}},
+        {"_Z21__spirv_WorkgroupId_xv", {"__dpcpp_nativecpu_get_wg_id", 0}},
+        {"_Z21__spirv_WorkgroupId_yv", {"__dpcpp_nativecpu_get_wg_id", 1}},
+        {"_Z21__spirv_WorkgroupId_zv", {"__dpcpp_nativecpu_get_wg_id", 2}}};
 
 Function *getReplaceFunc(Module &M, Type *T, StringRef Name) {
```

Comment:
I know it's not modified in this particular PR, but are we planning to use `T` in the future? Otherwise, should we get rid of it?

Reply:
Good point, that argument was left over from a while ago, I removed it, thank you